Yank/restraints.py

```diff
@@ -766,89 +1168,367 @@ class Boresch(OrientationDependentRestraint):
     [2] Mobley DL, Chodera JD, and Dill KA. J Chem Phys 125:084902, 2006.
         https://dx.doi.org/10.1063%2F1.2221683
 
+    Examples
+    --------
+    Create the ThermodynamicState.
+
+    >>> from openmmtools import testsystems, states
+    >>> system_container = testsystems.LysozymeImplicit()
+    >>> system, positions = system_container.system, system_container.positions
+    >>> thermodynamic_state = states.ThermodynamicState(system, 298*unit.kelvin)
+    >>> sampler_state = states.SamplerState(positions)
+
+    Identify ligand atoms. Topography automatically identify receptor atoms too.
+
+    >>> from yank.yank import Topography
+    >>> topography = Topography(system_container.topology, ligand_atoms=range(2603, 2621))
+
+    Create a partially defined restraint
+
+    >>> restraint = Boresch(restrained_atoms=[1335, 1339, 1397, 2609, 2607, 2606],
+    ...                     K_r=20.0*unit.kilocalories_per_mole/unit.angstrom**2,
+    ...                     r_aA0=0.35*unit.nanometer)
+
+    and automatically identify the other parameters. When trying to impose
+    a restraint with undefined parameters, RestraintParameterError is raised.
+
+    >>> try:
+    ...     restraint.restrain_state(thermodynamic_state)
+    ... except RestraintParameterError:
+    ...     print('There are undefined parameters. Choosing restraint parameters automatically.')
+    ...     restraint.determine_missing_parameters(thermodynamic_state, sampler_state, topography)
+    ...     restraint.restrain_state(thermodynamic_state)
+    ...
+    There are undefined parameters. Choosing restraint parameters automatically.
+
+    Get standard state correction.
+
+    >>> correction = restraint.get_standard_state_correction(thermodynamic_state)
+
     """
+    def __init__(self, restrained_atoms=None,
```

Comment:
Not if the user specify the atoms manually. We could figure out a way to do that by inspecting the bonds in the `ThermodynamicState` system when `restrain_state(thermodynamic_state)` is called, and raising an error if the 6 atoms are connected (directly or indirectly) by bonds.

In the near future though, all the atom selections that the user needs to make will probably be easily handled by the `Topography` object with DSL and/or SMIRKS specification, so it may not be worth it.